### PR TITLE
Apply bed leveling matrix when switching extruders

### PR DESCRIPTION
**Pull request by thinkyhead**
As noted in #122 by @jminardi, the auto bed leveling matrix is not taken into account when switching extruders. This patch applies the corrected difference between extruders.

### CODE REVIEW

**Comment by jminardi** [comment]
:+1:

**Comment by Roxy-3DPrintBoard** [comment]
I'm trying to get my Dual Extruder machine going...   So I'm going to be looking at this issue first hand!   I am having so many troubles getting Slicer to put out the right GCodes!

**Comment by Roxy-3DPrintBoard** [comment]
Has anybody tried these changes on a Delta ?   I'm starting to look at this issue.   And I'm not sure that these changes are going to handle what I'm seeing.

**Comment by Roxy-3DPrintBoard** [comment]
@AnHardt & @Thinkyhead  :   I'm fighting some horrible problems on my Delta.   I really need to get the bed leveled better and I may need to reprint the effector with some changes so I can get the nozzles at exactly the right height.   But I wanted to get you two thinking about this bug fix.

I haven't given this enough thought to know this for sure.  But I think this bug fix is only reasonable for Cartesian printers.   What I'm thinking is for Delta printers what has to happen is the  bed_level[][] matrix needs to be queried for the currently extrapolated offset based on which extruder is active.   Just because you switch extruders doesn't mean the effector should jump up or down.  And that is happening right now.

File: Marlin_main.cpp [hunk 1]
```diff
@@ -5625,9 +5625,24 @@ inline void gcode_T(uint8_t tmp_extruder) {
...
+          #if ENABLED(AUTO_BED_LEVELING_FEATURE)
+            // Offset extruder, make sure to apply the bed level rotation matrix
+            vector_3 tmp_offset_vec = vector_3(extruder_offset[X_AXIS][tmp_extruder],
+                                               extruder_offset[Y_AXIS][tmp_extruder],
+                                               extruder_offset[Z_AXIS][tmp_extruder]),
+                     act_offset_vec = vector_3(extruder_offset[X_AXIS][active_extruder],
+                                               extruder_offset[Y_AXIS][active_extruder],
+                                               extruder_offset[Z_AXIS][active_extruder]),
+                     offset_vec = tmp_offset_vec - act_offset_vec;
+            offset_vec.apply_rotation(plan_bed_level_matrix);
```
Comment (jminardi) [inline Marlin_main.cpp]: I just realized I pasted the wrong code when demonstrating this fix.

The `plan_bed_level_matrix` needs to be transposed here before applying the rotation.

See here: https://github.com/Voxel8/Marlin/blob/master/Marlin/Marlin_main.cpp#L6146

@Roxy-3DPrintBoard, maybe this is the issue you are seeing?

**Comment by Roxy-3DPrintBoard** [comment]
@jminardi 
> The plan_bed_level_matrix needs to be transposed here before applying the rotation.
> See here: https://github.com/Voxel8/Marlin/blob/master/Marlin/Marlin_main.cpp#L6146
> @Roxy-3DPrintBoard, maybe this is the issue you are seeing?

No.  I don't think this is the issue I'm seeing.   The Delta printers do not do a 'Tilted Bed' correction.   Instead, it is more like the mesh leveling where it looks at the table of values and interpolates what to do to flatten the movement.

I want to caution you that I have not looked at this enough to know for sure, but I think this code should not be part of what Delta's do.  I think we need to correct based on where the effector is and not where the nozzle is.  But like I say...  I need to dig in more on this so I'm more than willing to be corrected if something I'm saying doesn't make sense!   :)

**Comment by thinkyhead** [comment]
Ok, to make this fix properly, according to @jminardi just one change needs to be done. The line which reads:
```cpp
offset_vec.apply_rotation(plan_bed_level_matrix);
```
…should be changed to…
```cpp
offset_vec.apply_rotation(plan_bed_level_matrix.transpose(plan_bed_level_matrix));
```
I will make a new set of PRs for `dev` and `RCBugFix` that make this change as specified.

**Pull request closed without merging**